### ISSUE TITLE
[Support]: FFMPEG process crashed unexpectedly (and other errors)

### ISSUE THREAD

**Issue by TheGroundZero** [blakeblackshear/frigate#2286]
### Describe the problem you are having

Fresh setup of Frigate with a Reolink RLC-410 camera.
Copied the config of #1713 as mentioned in https://docs.frigate.video/configuration/camera_specific#reolink-410520-possibly-others
Frigate won't start.

Tested both streams in VLC and they work.

### Version

0.9.4-26ae608

### Frigate config file

```yaml
---
# Frigate
#
# https://docs.frigate.video/
# https://www.youtube.com/watch?v=pqDCEZSVeRk
#
mqtt:
  host: <MY HOST>
  user: <MY USER>
  password: <MY PASSWORD>

detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu

cameras:
  carport:
    ffmpeg:
      hwaccel_args:
        - -hwaccel
        - vaapi
        - -hwaccel_device
        - /dev/dri/renderD128
        - -hwaccel_output_format
        - yuv420p
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"
      inputs:
        - path: http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<USER>&password=<PASSWORD>
          roles:
            - record
            - rtmp
        - path: http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<USER>&password=<PASSWORD>
          roles:
            - detect
      output_args:
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
    detect:
      width: 896
      height: 672
      fps: 6
```

### Relevant log output

```shell
[2021-11-19 17:14:10] frigate.app                    INFO    : Starting Frigate (0.9.4-26ae608)
[2021-11-19 17:14:10] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2021-11-19 17:14:10] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2021-11-19 17:14:10] peewee_migrate                 INFO    : There is nothing to migrate
[2021-11-19 17:14:10] frigate.mqtt                   INFO    : MQTT connected
[2021-11-19 17:14:10] frigate.app                    INFO    : Output process started: 216
[2021-11-19 17:14:10] detector.cpu1                  INFO    : Starting detection process: 213
[2021-11-19 17:14:10] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
[2021-11-19 17:14:10] ws4py                          INFO    : Using epoll
[2021-11-19 17:14:10] frigate.app                    INFO    : Camera processor started for carport: 219
[2021-11-19 17:14:10] detector.cpu2                  INFO    : Starting detection process: 214
[2021-11-19 17:14:10] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
[2021-11-19 17:14:10] frigate.app                    INFO    : Capture process started for carport: 223
[2021-11-19 17:14:10] ws4py                          INFO    : Using epoll
[2021-11-19 17:14:15] frigate.video                  INFO    : carport: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2021-11-19 17:14:15] frigate.video                  INFO    : carport: ffmpeg process is not running. exiting capture thread...
[2021-11-19 17:14:30] watchdog.carport               ERROR   : FFMPEG process crashed unexpectedly for carport.
[2021-11-19 17:14:30] watchdog.carport               ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2021-11-19 17:14:30] watchdog.carport               ERROR   : You may have invalid args defined for this camera.
[2021-11-19 17:14:30] ffmpeg.carport.detect          ERROR   : [AVHWDeviceContext @ 0x55e15a97d100] No VA display found for device /dev/dri/renderD128.
[2021-11-19 17:14:30] ffmpeg.carport.detect          ERROR   : Device creation failed: -22.
[2021-11-19 17:14:30] ffmpeg.carport.detect          ERROR   : [h264 @ 0x55e15a96abc0] No device available for decoder: device type vaapi needed for codec h264.
[2021-11-19 17:14:30] ffmpeg.carport.detect          ERROR   : Device setup failed for decoder on input stream #0:0 : Invalid argument
[2021-11-19 17:14:35] frigate.video                  INFO    : carport: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2021-11-19 17:14:35] frigate.video                  INFO    : carport: ffmpeg process is not running. exiting capture thread...
[2021-11-19 17:14:40] watchdog.carport               ERROR   : FFMPEG process crashed unexpectedly for carport.
[2021-11-19 17:14:40] watchdog.carport               ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2021-11-19 17:14:40] watchdog.carport               ERROR   : You may have invalid args defined for this camera.
[2021-11-19 17:14:40] ffmpeg.carport.detect          ERROR   : [AVHWDeviceContext @ 0x55bed9299000] No VA display found for device /dev/dri/renderD128.
[2021-11-19 17:14:40] ffmpeg.carport.detect          ERROR   : Device creation failed: -22.
[2021-11-19 17:14:40] ffmpeg.carport.detect          ERROR   : [h264 @ 0x55bed9298bc0] No device available for decoder: device type vaapi needed for codec h264.
[2021-11-19 17:14:40] ffmpeg.carport.detect          ERROR   : Device setup failed for decoder on input stream #0:0 : Invalid argument
```

### FFprobe output from your camera

```shell
$ ffprobe "http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<USER>&password=<PASSWORD>"
ffprobe version 4.1.8-0+deb10u1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --prefix=/usr --extra-version=0+deb10u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Input #0, flv, from 'http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<USER>&password=<PASSWORD>':
  Metadata:
    displayWidth    : 2560
    displayHeight   : 1920
  Duration: 00:00:00.00, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264 (High), yuv420p(progressive), 2560x1920, 12 fps, 12 tbr, 1k tbn
    Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp



ffprobe "http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<USER>&password=<PASSWORD>"
ffprobe version 4.1.8-0+deb10u1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --prefix=/usr --extra-version=0+deb10u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Input #0, flv, from 'http://192.168.60.51/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<USER>&password=<PASSWORD>':
  Metadata:
    displayWidth    : 896
    displayHeight   : 672
  Duration: 00:00:00.00, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264 (High), yuv420p(progressive), 896x672, 12 fps, 12 tbr, 1k tbn
    Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Reolink RLC-410 5MP

**Comment by TheGroundZero** [comment]
Looks like issue is related to `hwaccel_args`.
I commented the section out and now it works.

FYI: My HA OS is installed in a Proxmox VM.
Server has a Intel(R) Xeon(R) CPU E5-2620 0 @ 2.00GHz

Would enabling `hv-evmcs` in Proxmox help?
![image](https://user-images.githubusercontent.com/2406013/142660842-2c28436e-a511-4e19-89f9-1b065ff68459.png)

**Comment by Sangoku** [comment]
> Looks like issue is related to `hwaccel_args`. I commented the section out and now it works.
> 
> FYI: My HA OS is installed in a Proxmox VM. Server has a Intel(R) Xeon(R) CPU E5-2620 0 @ 2.00GHz Currently no Coral stick (it's in backorder)
> 
> EDIT: Looks like I needed to use the "Intel-based CPUs (>=10th Generation) via Quicksync" config
> 
> ```yaml
> hwaccel_args:
>   - -hwaccel
>   - qsv
>   - -qsv_device
>   - /dev/dri/renderD128
> ```
> 
> Are there any other configs that would be relevant for my setup?

Omg thank you so much for this config. i replaced my on my
Intel NUC BOXNUC7I3BNH 
Intel Core 3-7100U
And it works like a charm! thank you! 


Posting my processor if somone else needs this gem! :D thank you. 

It worked fine untill HAssio released 2023.10.3 upwards.. 
After that they probably didnt support the old config anymre or no idea what..
